packages/eas-cli/src/project/discourageExpoGoForProd.ts

```diff
@@ -0,0 +1,73 @@
+import { Workflow } from '@expo/eas-build-job';
+import chalk from 'chalk';
+import getenv from 'getenv';
+import resolveFrom from 'resolve-from';
+
+import { resolveWorkflowPerPlatformAsync } from './workflow';
+import Log, { learnMore } from '../log';
+import type { ProfileData } from '../utils/profiles';
+import type { Client } from '../vcs/vcs';
+
+const suppressionEnvVarName = 'EAS_BUILD_NO_EXPO_GO_WARNING';
+
+export const discourageExpoGoForProd = (
+  buildProfiles: ProfileData<'build'>[] | undefined,
+  projectDir: string,
+  vcsClient: Client
+): void => {
+  detectExpoGoProdBuildAsync(buildProfiles, projectDir, vcsClient)
+    .then(usesExpoGo => {
+      if (usesExpoGo) {
```

Comment:
If you do `if (!usesExpoGo)` code will be indented a bit less and, in my opinion, more readable.